ros2param/test/test_cli.py

```diff
@@ -0,0 +1,718 @@
+                ExecuteProcess(
+                    cmd=['ros2', 'daemon', 'start'],
+                    name='daemon-start',
+                    on_exit=[
+                        # Add test fixture actions.
+                        Node(
+                            node_executable=sys.executable,
+                            arguments=[path_to_param_node_script],
+                            node_name='param_node',
+                            additional_env=additional_env
+                        ),
+                        Node(
+                            node_executable=sys.executable,
+                            arguments=[path_to_param_node_script],
+                            node_name='_hidden_param_node',
+                            additional_env=additional_env
+                        ),
+                        OpaqueFunction(function=lambda context: ready_fn())
+                    ],
+                    additional_env=additional_env
+                )
+            ]
+        ),
+    ])
+
+
+class TestROS2ParamCLI(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(
+        cls,
+        launch_service,
+        proc_info,
+        proc_output,
+        rmw_implementation
+    ):
+        @contextlib.contextmanager
+        def launch_node_command(self, arguments):
+            node_command_action = ExecuteProcess(
+                cmd=['ros2', 'param', *arguments],
+                additional_env={
+                    'RMW_IMPLEMENTATION': rmw_implementation,
+                    'PYTHONUNBUFFERED': '1'
+                },
+                name='ros2param-cli',
+                output='screen'
+            )
+            with launch_testing.tools.launch_process(
+                launch_service, node_command_action, proc_info, proc_output,
+                output_filter=launch_testing_ros.tools.basic_output_filter(
+                    # ignore launch_ros and ros2cli daemon nodes
+                    filtered_patterns=['.*ros2cli.*'],
+                    filtered_rmw_implementation=rmw_implementation
+                )
+            ) as node_command:
+                yield node_command
+        cls.launch_node_command = launch_node_command
+
+    # Set verb tests
+    @launch_testing.markers.retry_on_failure(times=3)
+    def test_set_invalid_param_existing_node(self):
+        with self.launch_node_command(
+                arguments=['set', '/param_node', 'unexistent_parameter', '1']) as node_command:
+            assert node_command.wait_for_shutdown(timeout=5)
+        assert node_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=[
+                "Setting parameter failed: ('Invalid access to undeclared parameter(s)', [])"],
+            text=node_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=3)
+    def test_set_valid_param_existing_node(self):
```

Comment:
@BMarchi I think we should probably test `ros2 param set` on each parameter type.